plutus-contract/src/Plutus/Contract/StateMachine/OnChain.hs

```diff
@@ -65,18 +64,23 @@ data StateMachine s i = StateMachine {
 
       -- | The 'AssetClass' of the thread token that identifies the contract
       --   instance.
-      smThreadToken :: Maybe AssetClass
+      smThreadToken :: Maybe TT.ThreadToken
```

Comment:
The comment could say that the initial step creates the thread token (if a thread token is defined)